File: kotlin-quarkus/src/main/kotlin/dev/sourcecraft/dolgintsev/entity/Product.kt
package dev.sourcecraft.dolgintsev.entity

import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import java.math.BigDecimal

@Entity
@Table(name = "products")
class Product : PanacheEntity() {
    companion object : PanacheCompanion<Product>

    @Column(nullable = false)
    lateinit var name: String

    @Column(length = 1000)
    var description: String? = null

    @Column(nullable = false)
    var price: BigDecimal = BigDecimal.ZERO

    @Column(nullable = false)
    var quantity: Int = 0
}

File: golang-gin/internal/models/product.go
package models

import (
	"gorm.io/gorm"
)

type Product struct {
	gorm.Model
	Name        string  `json:"name" gorm:"not null"`
	Description *string `json:"description" gorm:"type:varchar(1000)"`
	Price       float64 `json:"price" gorm:"not null;type:decimal(10,2)"`
	Quantity    int     `json:"quantity" gorm:"not null;default:0"`
}
